src/client/src/orangehrmClaimPlugin/pages/submitClaim/SubmitClaim.vue

```diff
@@ -0,0 +1,190 @@
+      <oxd-text tag="h6" class="orangehrm-main-title">
+        {{ $t('claim.create_claim_request') }}
+      </oxd-text>
+
+      <oxd-divider />
+
+      <oxd-form ref="formRef" :loading="isLoading" @submit-valid="onSave">
+        <oxd-form-row>
+          <oxd-grid :cols="3" class="orangehrm-full-width-grid">
+            <oxd-grid-item>
+              <oxd-input-field
+                v-model="request.claimEventId"
+                type="select"
+                :rules="rules.event"
+                :options="claimEvents"
+                :label="$t('claim.event')"
+                required
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <oxd-input-field
+                v-model="request.currencyId"
+                type="select"
+                :rules="rules.currency"
+                :options="currencies"
+                :label="$t('general.currency')"
+                required
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+
+        <oxd-form-row>
+          <oxd-grid :cols="3" class="orangehrm-full-width-grid">
+            <oxd-grid-item class="--span-column-2">
+              <oxd-input-field
+                v-model="request.remarks"
+                type="textarea"
+                :label="$t('claim.remarks')"
+                :rules="rules.remarks"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button
+            display-type="ghost"
+            :label="$t('general.cancel')"
+            @click="onCancel"
+          />
+          <submit-button :label="$t('claim.create')" />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {
+  required,
+  shouldNotExceedCharLength,
+} from '@/core/util/validation/rules';
+import {APIService} from '@ohrm/core/util/services/api.service';
+import useForm from '@ohrm/core/util/composable/useForm';
+import {navigate} from '@ohrm/core/util/helper/navigation';
+
+const claimRequest = {
+  claimEventId: null,
+  currencyId: null,
+  remarks: null,
+};
+
+export default {
+  name: 'ClaimRequest',
+
+  props: {
+    currencies: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/claim/requests',
+    );
+    const {formRef, reset} = useForm();
+
+    return {
+      http,
+      reset,
+      formRef,
+    };
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      request: {...claimRequest},
+      rules: {
+        event: [required],
+        currency: [required],
+        remarks: [shouldNotExceedCharLength(1000)],
+      },
+      claimEvents: [],
+    };
+  },
+
+  computed: {
+    computedRequest() {
+      if (this.request.claimEventId && this.request.currencyId) {
+        return {
```

Comment:
Form submission will happen only one time but this computed property will re run for every update, i don't think we need a computed property. transform the value before being send